### DISCUSSION TITLE
Why is UseSymboliclinksIfPossible disabled inside Visual Studio?

### DISCUSSION THREAD

**Discussion by giovanni-bozzano** [dotnet/msbuild#9419] [Q&A] [answered]
In my projects, I've set all the properties that enable symbolic links creation in the MSBuild Copy task. That is:

- `CreateSymbolicLinksForCopyFilesToOutputDirectoryIfPossible`
- `CreateSymbolicLinksForCopyAdditionalFilesIfPossible`
- `CreateSymbolicLinksForCopyLocalIfPossible`
- `CreateSymbolicLinksForAdditionalFilesIfPossible`
- `CreateSymbolicLinksForPublishFilesIfPossible`

For some reason, some of them are set to false when `BuildingInsideVisualStudio` is true:
https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4704
https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4721
https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4906
https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L5974

The only comment above those lines does not explain the reason for this behavior and I wasn't able to retrieve any information about it.

Is there a specific reason why these properties are disabled in Visual Studio?

**Comment by rokonec** [comment, marked as answer]
Although it looks good on paper, symbolic links were causing too many hard-to-debug build and execution issues, so I guess VS team decided it does more harm than gain and make it opt-in.

Additionally, [copy-on-write](https://devblogs.microsoft.com/engineering-at-microsoft/dev-drive-is-now-available/), which is supposed to be released to GA soon, will make "usage of symbolic link for copy performance" deprecated.

But, I was not behind that decision so please take my explanation with a grain of salt.
